var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -576,54 +578,120 @@ def headers(self):
     def setup_build_tests(self):
         """Copy the build test files after the package is installed to an
         install test subdirectory for use during `spack test run`."""
-        self.cache_extra_test_sources('src/ksp/ksp/tutorials')
-        self.cache_extra_test_sources('src/snes/tutorials')
+        self.cache_extra_test_sources([
+            join_path('src', 'ksp', 'ksp', 'tutorials'),
+            join_path('src', 'snes', 'tutorials')
+        ])
+
+    def run_ex50_test(self, runexe, runopt, w_dir, makefile_dir):
+        """Run stand alone test: ex50"""
+
+        self.run_test('make',
+                      options=['-f', makefile_dir, 'ex50'],
+                      purpose='test: compile ex50',
+                      work_dir=w_dir)
+
+        testexe = ['ex50', '-da_grid_x', '4', '-da_grid_y', '4']
+        testdict = {
+            None: [],
+            '+superlu-dist':
+            ['-pc_type', 'lu', '-pc_factor_mat_solver_type', 'superlu_dist'],
+            '+mumps':
+            ['-pc_type', 'lu', '-pc_factor_mat_solver_type', 'mumps'],
+            '+hypre':
+            ['-pc_type', 'hypre', '-pc_hypre_type', 'boomeramg'],
+            '+mkl-pardiso':
+            ['-pc_type', 'lu', '-pc_factor_mat_solver_type', 'mkl_pardiso'],
+        }
+
+        for feature, featureopt in testdict.items():
+            if not feature or feature in self.spec:
+                purpose_str = 'test: run ex50 example with {0}'.format(feature)
+                self.run_test(runexe,
+                              options=runopt + testexe + featureopt,
+                              purpose=purpose_str,
+                              work_dir=w_dir)
+
+    def run_ex7_test(self, runexe, runopt, w_dir, makefile_dir):
+        """Run stand alone test: ex7 with cuda"""
+
+        self.run_test('make',
+                      options=['-f', makefile_dir, 'ex7'],
```

Comment:
can we remove the usage of `makefile_dir` completely and simplify this code? 

`make` should default to using `makefile` in `work_dir=w_dir`